JOSE_paper/paper.md

```diff
@@ -0,0 +1,223 @@
+in their work.
+It is designed to introduce key tools and techniques to facilitate the writing of
+better code, but also to raise awareness of research software engineering and provide pointers
+to allow researchers to continue exploring and learning beyond this material.
+
+
+
+# Learning objectives
+
+The key learning objective for the workshop is to _Introduce key tools and concepts of
+research software engineering, and how they can be applied in everyday use to write
+higher-quality code_.
+
+More specifically we cover:
+
+* Why software engineering principles matter
+* (Virtual) environments and dependencies
+* Code standards and best practice (through PEP)
+  * Code formatters and linting/static-analysis tools
+* Documentation
+  * Docstrings
+  * Advice on commenting
+  * The idea of self-documenting code and variable naming
+* General principles for better code
+  * Magic Numbers
+  * Removal of hard-coded content to config files
+  * use of f-strings in Python.
+
+
+
+# Teaching materials
+
+All of the teaching materials for this course are available online in a GitHub repository.
+
+
+## Slides
+
+All of the key material is contained in a slide deck for the course available online
+and linked from the repository README.
+The slides are written in [Quarto](https://quarto.org/) markdown [@Allaire_Quarto_2022]
+and rendered as [reveal.js](https://revealjs.com/) html.
+Source and instructions on how to render are included in the repository should
+others wish to tailor them to their specifications.
+
+The slides are broken into separate files by section covering:
+
+- Introduction to research software engineering, why it matters, and some examples,
+- The use of virtual environments and specifying project dependencies,
+- Code formatting for consistent readable layout,
+- Naming principles for readable, re-useable code,
+- Linting and static analysis of code,
+- Writing docstrings and comments, and
+- Other general principles including magic numbers, avoiding hardcoded values, etc.
+
+This makes it easy to remove sections to tailor the course, or for additional topics to be added in future.
+
+## Exercises
+
+The slides are interspersed with a series of exercises, each one applying techniques
+that have just been taught.
+
+We set up a scenario where users have been provided a working, but poorly written piece
+of code that they are now expected to re-use in their work.
+As they progress through the exercises they apply their new skills to transform this
+into a much more readable and understandable piece of code.
+
+The script we use is based on some of the code and data available in @Irving2019.
+This was chosen as it features typical applications we expect users to encounter
+(reading from a dataset, processing, and plotting).
+The course was originally designed to be taught to geoscientists, and then at a climate
+science summer school, and we wanted participants to be focussed on the changes to the
+code rather than worrying about 
```

Comment:
worrying about its semantics?

Reply:
Done in https://github.com/jatkinson1000/power-up-python/pull/14/commits/2816d304cbae61641b9a2d4d75ea7fc8b2954b4d